kr_mesh_visualization/src/mesh_visualization.cpp

```diff
@@ -1,83 +1,100 @@
-  marker.ns = ros::this_node::getName();
-  marker.id = 0;
-  marker.type = visualization_msgs::Marker::MESH_RESOURCE;
-  marker.action = visualization_msgs::Marker::ADD;
-  marker.pose = pose;
-  marker.scale.x = scale_x;
-  marker.scale.y = scale_y;
-  marker.scale.z = scale_z;
-  marker.color.a = color_a;
-  marker.color.r = color_r;
-  marker.color.g = color_g;
-  marker.color.b = color_b;
-  marker.mesh_resource = mesh_resource;
-  pub_vis.publish(marker);
-}
+ public:
+  MeshVisualizationNode() : Node("kr_mesh_visualization")
+  {
+    this->declare_parameter<std::string>("mesh_resource",
+                                         std::string("package://kr_mesh_visualization/mesh/hummingbird.mesh"));
+    this->declare_parameter<double>("color/r", 1.0);
+    this->declare_parameter<double>("color/g", 0.0);
+    this->declare_parameter<double>("color/b", 0.0);
+    this->declare_parameter<double>("color/a", 1.0);
+    this->declare_parameter<double>("scale/x", 1.0);
+    this->declare_parameter<double>("scale/y", 1.0);
+    this->declare_parameter<double>("scale/z", 1.0);
+    this->declare_parameter<std::string>("new_frame_id", std::string(""));
 
-static void any_callback(const topic_tools::ShapeShifter::ConstPtr &msg)
-{
-  // ROS_INFO_STREAM("Type: " << msg->getDataType());
+    this->get_parameter("mesh_resource", mesh_resource);
+    this->get_parameter("color/r", color_r);
+    this->get_parameter("color/g", color_g);
+    this->get_parameter("color/b", color_b);
+    this->get_parameter("color/a", color_a);
+    this->get_parameter("scale/x", scale_x);
+    this->get_parameter("scale/y", scale_y);
+    this->get_parameter("scale/z", scale_z);
+    this->get_parameter("new_frame_id", new_frame_id);
 
-  if(msg->getDataType() == "geometry_msgs/PoseStamped")
+    using namespace std::placeholders;
+    pub_vis_ = this->create_publisher<visualization_msgs::msg::Marker>("~/robot", 10);
+    sub_odom_ = this->create_subscription<nav_msgs::msg::Odometry>("~/input", 10,
+                std::bind(&MeshVisualizationNode::odom_callback, this, _1));
+    
+
+    /* Uncomment any of the below subscribers to support another message type */
+    // sub_pwcs_ = this->create_subscription<geometry_msgs::msg::PoseWithCovarianceStamped>(
+    //     "~/input", 10, std::bind(&MeshVisualizationNode::callback_pose_with_covariance_stamped, this, _1));
+    // sub_ps_ = this->create_subscription<geometry_msgs::msg::PoseStamped>(
+    //     "~/input", 10, std::bind(&MeshVisualizationNode::callback_pose_stamped, this, _1));
+  }
+
+ protected:
+  std::string mesh_resource, new_frame_id;
+  rclcpp::Publisher<visualization_msgs::msg::Marker>::SharedPtr pub_vis_;
+  rclcpp::Subscription<geometry_msgs::msg::PoseStamped>::SharedPtr sub_ps_;
+  rclcpp::Subscription<geometry_msgs::msg::PoseWithCovarianceStamped>::SharedPtr sub_pwcs_;
+  rclcpp::Subscription<nav_msgs::msg::Odometry>::SharedPtr sub_odom_;
+  double color_r, color_g, color_b, color_a;
```

Comment:
These can be a vector

Reply:
Yes, but just maintained the same format from the original code where each color and scale was separately defined.